xtask/src/bootstrap_blake3_groth16.rs

```diff
@@ -0,0 +1,219 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is automatically generated by:
+// cargo xtask bootstrap-groth16
+
+"#;
+
+const SOLIDITY_VERIFIER_TARGET: &str = "contracts/src/bitvm/Groth16Verifier.sol";
+const SOLIDITY_CONTROL_ID_PATH: &str = "contracts/src/bitvm/ControlID.sol";
+const SOLIDITY_TEST_RECEIPT_PATH: &str = "contracts/test/Blake3TestReceipt.sol";
+const RUST_VERIFIER_PATH: &str = "risc0/groth16/src/verifier/blake3.rs";
+
+impl BootstrapBlake3Groth16 {
+    pub fn run(&self) {
+        let current_dir = std::env::current_dir().unwrap();
+        let blake3_setup_path = current_dir.join(&self.blake3_groth16_setup_dir);
+        let risc0_ethereum_path = current_dir.join(&self.risc0_ethereum_path);
+        if !self.test_receipt_only {
+            bootstrap_verifying_key(&blake3_setup_path, &risc0_ethereum_path);
+            bootstrap_control_id(&risc0_ethereum_path);
+        }
+        bootstrap_test_receipt(&risc0_ethereum_path);
+    }
+}
+
+fn bootstrap_verifying_key(blake3_setup_path: &Path, risc0_ethereum_path: &Path) {
+    let solidity_verifier_target = risc0_ethereum_path.join(SOLIDITY_VERIFIER_TARGET);
+    let solidity_verifier_source = blake3_setup_path.join("verifier.sol");
+    std::fs::copy(&solidity_verifier_source, solidity_verifier_target).unwrap();
+    let solidity_code = fs::read_to_string(solidity_verifier_source).unwrap();
+    let mut rust_code = fs::read_to_string(RUST_VERIFIER_PATH).unwrap();
+
+    let solidity_constants = [
+        "alphax", "alphay", "betax1", "betax2", "betay1", "betay2", "gammax1", "gammax2",
+        "gammay1", "gammay2", "deltax1", "deltax2", "deltay1", "deltay2", "IC0x", "IC0y", "IC1x",
+        "IC1y",
+    ];
+
+    let rust_constants = [
+        "ALPHA_X", "ALPHA_Y", "BETA_X1", "BETA_X2", "BETA_Y1", "BETA_Y2", "GAMMA_X1", "GAMMA_X2",
+        "GAMMA_Y1", "GAMMA_Y2", "DELTA_X1", "DELTA_X2", "DELTA_Y1", "DELTA_Y2", "IC0_X", "IC0_Y",
+        "IC1_X", "IC1_Y",
+    ];
+
+    for (i, constant) in solidity_constants.into_iter().enumerate() {
+        let re = Regex::new(&format!(r"uint256 constant\s+{constant}\s*=\s*(\d+);")).unwrap();
+        if let Some(caps) = re.captures(&solidity_code) {
+            let rust_re = Regex::new(&format!(
+                "const {}: &str =[\\r\\n\\s]*\"\\d+\";",
+                rust_constants[i]
+            ))
+            .unwrap();
+            rust_code = rust_re
+                .replace(
+                    &rust_code,
+                    &format!("const {}: &str = \"{}\";", rust_constants[i], &caps[1]),
+                )
+                .to_string();
```

Comment:
oof, I guess this works, if it was me I would use `syn` which contains a Rust parser.